starlite/controller/generic_controller.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+from typing_extensions import Self, get_args, get_origin
+
+from .base import Controller
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+__all__ = ("GenericController",)
+
+T = TypeVar("T")
+
+
+class GenericController(Controller, Generic[T]):
+    """Controller type that supports generic inheritance hierarchies."""
+
+    _data_type: type[T]
+
+    def __class_getitem__(cls, data_type: type[T]) -> type[Self]:
+        return type(f"GenericController[{data_type.__name__}]", (cls,), {"_data_type": data_type})
+
+    def _get_new_args(self, args: tuple[Any, ...], origin: Any | None) -> tuple[Any, ...] | None:
+        if not args and origin:
+            return None
+
+        type_var_found = False
+        new_args = []
+        for arg in args:
+            if isinstance(arg, TypeVar):
+                new_args.append(self._data_type)
+                type_var_found = True
+                continue
+            new_args.append(arg)
+
+        if not type_var_found:
+            return None
```

Comment:
I think we should raise an ImproperlyConfiguredException - generic controllers should be used with geberics

Reply:
This is called for every parameter to the handler though - so that would break something like:

```py
class GenCont(...):
   @put()
   def handle_put(item_id: int, generic_collection: list[ThingT]) -> ...:
       ...
```
The method is for identifying that `ThingT` needs to be swapped with a concrete `Thing` in the signature model parameter annotation, and the `return None` is for breaking early when we detect the `item_id` case - that is, where there is no chance of there being a type var in that parameter's annotation.